harness/determined/pytorch/_callback.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict
+
+import torch
+
+
+class PyTorchCallback:
+    """
+    Abstract base class used to define a callback that should execute during
+    the lifetime of a PyTorchTrial.
+
+    .. warning::
+        If you are defining a stateful callback (e.g. it mutates a self
+        attribute over it's lifetime), you must also override state_dict() and
+        load_state_dict() to ensure this state can be serialized and deserialized
+        over checkpoints.
+
+    .. warning::
+        If distributed or parallel training is enabled, every shard will
```

Comment:
In our other docs, I don't believe we use the term `shard`, for consistency might be better change it to GPU.